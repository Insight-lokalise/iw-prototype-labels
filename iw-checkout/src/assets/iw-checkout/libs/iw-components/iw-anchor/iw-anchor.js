import React from 'react'

/**
 * A wrapper for an <a> tag.
 *
 * Do we want to enforce any properties on an Anchor?
 * Do we want to apply any global styles?
 *
 * Will later integrate with / be replaced by react-router's Link
 *
 * @param {Object} props Any normal property you would apply to an <a>.
 */
export function IWAnchor(props) {
    return (
        <a {...props} >
            { props.children }
        </a>
    )
}
