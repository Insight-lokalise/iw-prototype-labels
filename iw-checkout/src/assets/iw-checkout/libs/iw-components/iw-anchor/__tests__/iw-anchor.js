import React from 'react'
import renderer from 'react-test-renderer'
import { IWAnchor } from './../iw-anchor'

describe('<IWAnchor>', () => {
    it('should render children', () => {
        const component = renderer.create(
            <IWAnchor className='B__E--M'>
                Clickable
                <strong>text</strong>
            </IWAnchor>
        )
        expect(component).toMatchSnapshot()
    })
})
