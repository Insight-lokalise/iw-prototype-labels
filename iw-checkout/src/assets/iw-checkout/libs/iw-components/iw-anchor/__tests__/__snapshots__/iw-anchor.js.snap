// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IWAnchor> should render children 1`] = `
<a
  className="B__E--M"
>
  Clickable
  <strong>
    text
  </strong>
</a>
`;
